tests/integration-tests/tests/api/test_api_client.py

```diff
@@ -136,3 +142,62 @@ def _test_delete_cluster(client, cluster, region):
     response = client.list_clusters(region=region)
     cluster_names = [c["cluster_name"] for c in response["items"]]
     assert_that(cluster_names).does_not_contain(cluster_name)
+
+
+@pytest.mark.usefixtures("instance")
+def test_custom_image(request, region, os, pcluster_config_reader, api_client):
+    base_ami = retrieve_latest_ami(region, os)
+
+    config_file = pcluster_config_reader(config_file="image.config.yaml", parent_image=base_ami)
+    with open(config_file) as config_file:
+        config = config_file.read()
+
+    stack_name = generate_stack_name("integ-tests", request.config.getoption("stackname_suffix"))
+    client = image_operations_api.ImageOperationsApi(api_client)
+
+    with soft_assertions():
+        resp = _test_build_image(client, config, region, stack_name)
+        LOGGER.info("Build image response: %s", resp)
+        image = resp["image"]
+        image_id = image["image_id"]
+        assert_that(image_id).is_equal_to(stack_name)
+
+        # CFN stack is deleted as soon as image is available
+        _cloudformation_wait(region, stack_name, "stack_delete_complete")
+
+        _test_describe_image(client, image, region)
+        _test_list_images(client, image, region)
```

Comment:
similar to cluster can we do a describe and a list also when the creation is in progress?